rp2-pio/pio.go

```diff
@@ -240,6 +235,46 @@ func (pio *PIO) SetInputSyncBypassMasked(bypassMask, pinMask uint32) {
 	pio.hw.INPUT_SYNC_BYPASS.ReplaceBits(bypassMask, pinMask, 0)
 }
 
+// SetNextPIOMask configures the 4-bit mask for state machines in the next PIO block
+// that should be affected by ClkDivRestart() and SetEnabled() functions on this PIO
+// block's state machines, allowing for cycle-perfect synchronization. RP2350-only.
+func (pio *PIO) SetNextPIOMask(mask uint8) {
+	pio.setNextPIOMask(uint32(mask & 0xF))
+}
+
+// SetPrevPIOMask configures the 4-bit mask for state machines in the previous PIO
+// block that should be affected by ClkDivRestart() and SetEnabled() functions on this
+// PIO block's state machines, allowing for cycle-perfect synchronization. RP2350-only.
+func (pio *PIO) SetPrevPIOMask(mask uint8) {
+	pio.setPrevPIOMask(uint32(mask & 0xF))
+}
```

Comment:
If these are RP2350 only then lets hide them behind a build tag. I suggest changing `rp2-pio/config-rp2040.go` and its rp2350 counterpart to simply: `rp2-pio/pio-rp2040.go` and  `rp2-pio/pio-rp2350.go` and storing all architecture specific logic there.